app/assets/javascripts/basetypes/Code.js

```diff
@@ -12,7 +12,7 @@ Code.prototype.cast = (code) => {
     throw new Error(`Code: ${code} does not have a codeSystem`);
   }
 
-  const val = { code: code.code, codeSystem: code.codeSystem };
+  const val = { code: code.code, codeSystem: code.codeSystem, system: code.codeSystem };
```

Comment:
As I was testing, I noticed that the Equivalent check in cqm-execution does not like comparing and "Object" with a "cql.Code".  instead of creating a general object here, create a cql.Code

const val = new cql.Code(code.code, code.codeSystem );
val.codeSystem = code.codeSystem;
return val2;

Not really related to here.  But i also noticed that a Coded value for an "Any" result is also being casted as an oject (not a Code, which also breaks equivalent).  Consider adding this to Any.js in this PR

if (any && any.code && any.codeSystem) {
    const casted = new cql.Code(any.code, any.codeSystem)
    casted.codeSystem = any.codeSystem
    return casted
  }

Reply:
How are you testing this? The code equivalency is not evaluating wrong in any of the tests when we are using code as Object. It should be fine using Object or cql.Code since the comparing attributes (code,codesystem) will be available on both Object and cql.Code types.

Reply:
This looks like this was addressing in cql-execution 1.2.1.  The js-ecqm-engine has been using 1.2.0.